### PR TITLE
Add more tests for buf.work.yaml and normalize and validate DirPaths

### PR DESCRIPTION
**Pull request by oliversun9**
This PR adds tests for `buf_work_yaml_file.go`.

This PR also fixes a bug where `newBufWorkYAMLFile` does not set `dirPaths` as the normalized input paths (but set `dirPaths` as the input paths as-is).

### CODE REVIEW

File: private/bufnew/bufconfig/buf_work_yaml_file.go [hunk 1]
```diff
@@ -200,52 +200,59 @@ func writeBufWorkYAMLFile(writer io.Writer, bufWorkYAMLFile BufWorkYAMLFile) err
...
 		}
-		dirPathMap[dirPath] = struct{}{}
+		normalDirPathToDirPath[normalDirPath] = dirPath
 	}
 	// We already know the paths are unique due to above validation.
 	// We sort to print deterministic errors.
-	if err := validateDirPathsNoOverlap(slicesextended.MapToSortedSlice(dirPathMap)); err != nil {
-		return err
+	// TODO: use this line:
+	// sortedNormalDirPaths := slicesextended.MapKeysToSortedSlice(normalDirPathToDirPath)
```
Comment (oliversun9) [inline private/bufnew/bufconfig/buf_work_yaml_file.go:226]: https://github.com/bufbuild/buf/pull/2612

File: private/bufnew/bufconfig/buf_work_yaml_file.go [hunk 2]
```diff
@@ -129,13 +129,13 @@ type bufWorkYAMLFile struct {
 }
 
 func newBufWorkYAMLFile(fileVersion FileVersion, dirPaths []string) (*bufWorkYAMLFile, error) {
-	if err := validateBufWorkYAMLDirPaths(dirPaths); err != nil {
+	sortedNormalDirPaths, err := validateBufWorkYAMLDirPaths(dirPaths)
```
Comment (bufdev) [inline private/bufnew/bufconfig/buf_work_yaml_file.go:132]: We've used `normalized` throughout rest of codebase, not `normal`, update

**Review by bufdev — COMMENTED** [review]
You'll also want to update the documentation for `DirPaths` to state that paths are normalized and validated when constructing a `BufWorkYAMLFile`, and that `DirPaths` returns only normalized and validated paths.

**Comment by oliversun9** [comment]
> 4ee426d

fixed in [4ee426d](https://github.com/bufbuild/buf/pull/2613/commits/4ee426de8c98ff996fe3cd808ca03ca5493e9a1c)

**Comment by oliversun9** [comment]
> You'll also want to update the documentation for `DirPaths` to state that paths are normalized and validated when constructing a `BufWorkYAMLFile`, and that `DirPaths` returns only normalized and validated paths.

fixed in [4ee426d](https://github.com/bufbuild/buf/pull/2613/commits/4ee426de8c98ff996fe3cd808ca03ca5493e9a1c)

File: private/bufnew/bufconfig/buf_work_yaml_file.go (hunk 2 above)
Comment (oliversun9) [reply private/bufnew/bufconfig/buf_work_yaml_file.go]: fixed in [f2481c7](https://github.com/bufbuild/buf/pull/2613/commits/f2481c785d23efba16fe8534c34d238e3f9256e8)

**Review by bufdev — APPROVED** [review]

**Pull request merged**